containers/pre_built_workers/cxx/Dockerfile

```diff
@@ -12,7 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM l.gcr.io/google/bazel:latest
+# See https://github.com/grpc/grpc/blob/master/tools/dockerfile/README.md
+FROM us-docker.pkg.dev/grpc-testing/testing-images-public/bazel:0f909e43012a80faa92e07b7871268841ce56ebc
```

Comment:
Since we have it, why not add the sha256 image digest for an extra sanity check?

`us-docker.pkg.dev/grpc-testing/testing-images-public/bazel:0f909e43012a80faa92e07b7871268841ce56ebc@sha256:1118150d9d9479787165fff49f660a3dc633f1c57604305460172fc1916aa022`

Reply:
Done.